stetho/src/main/java/com/facebook/stetho/inspector/network/NetworkEventReporter.java

```diff
@@ -25,10 +25,31 @@
  *                   `-----------------------------`--------> httpExchangeFailed
  * </pre>
  *
- * Note that {@link #interpretResponseStream} combined with {@link DefaultResponseHandler}
+ * <p>Note that {@link #interpretResponseStream} combined with {@link DefaultResponseHandler}
  * will automatically invoke {@link #dataReceived}, {@link #responseReadFailed} and
  * {@link #responseReadFinished}.  If you use your own custom {@link ResponseHandler} you
- * must be sure to invoke these methods manually.
+ * must be sure to invoke these methods manually.</p>
+ *
+ * <p>For arbitrary sockets or explicitly for WebSockets, the following call flow must be met:</p>
+ *
+ * <pre>
+ * webSocketCreated +---> webSocketWillSendHandshakeRequest ----> webSocketHandshakeResponseReceived
+ *                  |                                                              |
+ *                  |                     ,----------------------+-----------------+--------,
+ *                  |                     v                      v                          |
+ *                  +---------> [ webSocketFrameSent | webSocketFrameReceiver ] ---,        |
```

Comment:
If you have a logical grouping of frame sent and frame received there is no need to have a dotted line to and from each of them. A single inbound and outbound line would be sufficient